macro-fullcalendar-api/src/main/java/org/xwiki/fullcalendar/model/MoccaCalendarEvent.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.fullcalendar.model;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.xwiki.stability.Unstable;
+
+/**
+ * Represents the data of a mocca calendar event.
+ *
+ * @version $Id$
+ * @since 2.4.0
+ */
+@Unstable
+public class MoccaCalendarEvent
+{
+    private String id;
+
+    private String title;
+
+    private Date start;
+
+    private Date end;
+
+    private boolean allDay;
+
+    private String description;
+
+    private String location;
+
+    private String status;
+
+    private int recurrent;
+
+    private Date recEndDate;
+
+    private String recurrenceFreq;
+
+    private List<RecurrentEventModification> modificationList;
+
+    private String groupId;
+
+    /**
+     * Default constructor.
+     */
+    public MoccaCalendarEvent()
+    {
+        modificationList = new ArrayList<>();
+    }
+
+    /**
+     * Copy constructor.
+     *
+     * @param calendarEvent {@link MoccaCalendarEvent} to be copied.
+     */
+    public MoccaCalendarEvent(MoccaCalendarEvent calendarEvent)
+    {
+        this.setId(calendarEvent.getId());
+        this.setTitle(calendarEvent.getTitle());
+        this.setStart(calendarEvent.getStart());
+        this.setEnd(calendarEvent.getEnd());
+        this.setAllDay(calendarEvent.isAllDay());
+        this.setDescription(calendarEvent.getDescription());
+        this.setLocation(calendarEvent.getLocation());
+        this.setStatus(calendarEvent.getStatus());
+        this.setRecurrent(calendarEvent.isRecurrent());
+        this.setRecEndDate(calendarEvent.getRecEndDate());
+        this.setRecurrenceFreq(calendarEvent.getRecurrenceFreq());
+        this.setGroupId(calendarEvent.getGroupId());
+        modificationList = new ArrayList<>(calendarEvent.getModificationList());
+    }
+
+    /**
+     * Get group id.
```

Comment:
You could be a little more descriptive about the properties. If I was to use this API, I wouldn't know what to do with this property. You could say something like. "You can use this field to specify a relation between two or more events, such as the events generated from a common recurrence rule".

This can be applied to all the methods of this class.

To not duplicate the doc for the getters and setters, you can document only the setter and fill in the getter with something like:
```
/**
 * @returns see {@link #setParamName()}
 */